zerver/views/upload.py

```diff
@@ -84,3 +84,7 @@ def get_uploaded_file(request, realm_id_str, filename,
     # type: (HttpRequest, str, str, bool) -> HttpResponse
     user_profile = request.user
     return serve_file_backend(request, user_profile, realm_id_str, filename, redir)
+
+@has_request_variables
+def api_upload_file(request, user_profile):
+    return upload_file_backend(request, user_profile)
```

Comment:
This function seems unnecessary; you can just put `zerver.views.upload.upload_file_backend` in the urls.py file.